### PR TITLE
bpo-38962: Fix reference leak in new_interpreter()

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue38962

Automerge-Triggered-By: @pablogsal

### CODE REVIEW

File: Python/pylifecycle.c [hunk 1]
```diff
@@ -648,9 +648,11 @@ pycore_init_builtins(PyThreadState *tstate)
     if (interp->builtins_copy == NULL) {
         goto error;
     }
+    Py_DECREF(bimod);
     return _PyStatus_OK();
 
 error:
+    Py_DECREF(bimod);
```
Comment (vstinner) [inline Python/pylifecycle.c]: It should be Py_XDECREF for the if (bimod == NULL) { goto error; } case.

Comment (pablogsal) [reply Python/pylifecycle.c]: Will update soon

Comment (pablogsal) [reply Python/pylifecycle.c]: @vstinner Updated in d00de0e

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**